File: webapi/Models/Filter.cs
using System.Collections.Generic;
using System.Reflection;

namespace webapi.Models
{
    public record FilterData
    {
        public required string FieldName { get; set; }
        public required string Value { get; set; }
    }
    public record Filter
    {
        public required string FieldName { get; set; }
        public required string Value { get; set; }
    }
}

File: reactapp/src/models/Filter.ts
export interface FilterData {
    fieldName: string,
    value: string,
}

export interface Filter {
    fieldName: string,
    value: string,
}

export const defaultFilter: Filter = {
    fieldName: "",
    value: "",
}

export function createFilter(name: string, value: string):Filter {
    return {
        fieldName: name,
        value: value
    }
}